### PR TITLE
Alex/#11 keep log image requested in the db

### LINKED ISSUE #11: Model selection for images
We need the ability to select the model for image generation as well as keep a log of what was requested and generated at the user level in the DB

### PR DESCRIPTION
**Pull request by aleksandr-efimenko**
I implemented a log of what was requested and generated at the user level in the DB. Every time a user makes a request to the Open AI, related data (requestText, numberOfImages, imageSize, and userId) is stored in the DB. 

Initially, I wanted to store the URLs of generated images in the DB and receive them when a user opens the page. However, it turned out that images stored on the Open Ai server are available only for 1 hour. So for this feature, we need to consider storing images on our server.

Closes #11

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -83,3 +84,28 @@ model OpenAIModels {
...
+model GeneratedImages {
+    id                       String                  @id @default(cuid())
+    createdAt                DateTime                @default(now())
+    imageUrl                 String
+    ImageGenerationRequest   ImageGenerationRequest? @relation(fields: [imageGenerationRequestId], references: [id])
+    imageGenerationRequestId String?
+}
+
+model ImageSize {
+    size                   String                   @id
```
Comment (cohenoshriconsulting) [inline prisma/schema.prisma:109]: Add column size_id and use that as the relationship

File: src/hooks/saveImageRequest.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  //     const requests = getGeneratedImagesQuery.data?.map(
+  //       (d) => d.generatedImages
+  //     );
+  //     if (!requests) return;
+  //     const images = requests.flat().map((d) => d.imageUrl);
+  //     setGeneratedImages(images);
+  //   }, [getGeneratedImagesQuery.data, setGeneratedImages]);
+
+  const saveUserRequest = async ({
+    values,
```
Comment (cohenoshriconsulting) [inline src/hooks/saveImageRequest.ts:34]: Use plural for arrays

Comment (aleksandr-efimenko) [reply src/hooks/saveImageRequest.ts]: Renamed the object, use singular name

File: prisma/schema.prisma (hunk 1 above)
Comment (aleksandr-efimenko) [reply prisma/schema.prisma]: Removed the image size table, use string for image sizes in the requests table

**Review by cohenoshriconsulting — APPROVED** [review]

**Pull request merged**